bio2zarr/vcf.py

```diff
@@ -940,70 +981,108 @@ def convert_partition(
             f"{num_records} records"
         )
-        return partition_index, tcw.field_summaries, num_records
 
     @staticmethod
-    def convert(
-        vcfs, out_path, *, column_chunk_size=16, worker_processes=1, show_progress=False
-    ):
+    def convert_init(vcfs, out_path, *, num_partitions=1, worker_processes=1, show_progress=False):
         out_path = pathlib.Path(out_path)
         # TODO make scan work in parallel using general progress code too
-        target_num_partitions = max(1, worker_processes * 4)
         vcf_metadata, header = scan_vcfs(
             vcfs,
             worker_processes=worker_processes,
             show_progress=show_progress,
-            target_num_partitions=target_num_partitions,
+            target_num_partitions=num_partitions,
         )
         pcvcf = PickleChunkedVcf(out_path, vcf_metadata, header)
         pcvcf.mkdirs()
 
-        logger.info(
-            f"Exploding {pcvcf.num_columns} columns {vcf_metadata.num_records} variants "
-            f"{pcvcf.num_samples} samples"
-        )
+        pcvcf.write_metadata()
+        pcvcf.write_header()
+        return pcvcf
+
+    def convert_slice(self, start, stop, *, worker_processes=1, show_progress=False, column_chunk_size=16):
+        if start < 0:
+            raise ValueError(f"start={start} must be non-negative")
+        if stop > self.num_partitions:
+            raise ValueError(f"stop={stop} must be less than the number of partitions")
+        if start == 0 and stop == self.num_partitions:
+            num_records_to_process = self.num_records
+            logger.info(
+                f"Exploding {self.num_columns} columns {self.metadata.num_records} variants "
+                f"{self.num_samples} samples"
+            )
+        else:
+            num_records_to_process = None
+            logger.info(
+                f"Exploding {self.num_columns} columns {self.num_samples} samples"
+                f" from partitions {start} to {stop}"
+            )
+
         progress_config = core.ProgressConfig(
-            total=vcf_metadata.num_records,
+            total=num_records_to_process,
             units="vars",
             title="Explode",
             show=show_progress,
         )
-        with core.ParallelWorkManager(worker_processes, progress_config) as pwm:
-            for j, partition in enumerate(vcf_metadata.partitions):
-                pwm.submit(
-                    PickleChunkedVcf.convert_partition,
-                    vcf_metadata,
-                    j,
-                    out_path,
-                    column_chunk_size=column_chunk_size,
-                )
-            num_records = 0
-            partition_summaries = []
-            for index, summary, num_records in pwm.results_as_completed():
-                partition_summaries.append(summary)
-                vcf_metadata.partitions[index].num_records = num_records
-
+        if stop-start == 1:
```

Comment:
I'd rather just use the same code path, if only for progress tracking reasons. It's not worth the complexity to keep this case a bit quicker.